### PR TITLE
Another wizard walks into the bar

### PR DESCRIPTION
**Pull request by darzu**

### CODE REVIEW

File: src/game/bullet-collision.ts [hunk 1]
```diff
@@ -39,11 +40,7 @@ export function registerBulletCollisionSystem(em: EntityManager) {
           );
           for (let otherBullet of otherBullets) {
             if (otherBullet) {
-              resources.detectedEvents.raise({
-                type: "bullet-bullet",
-                entities: [o.id, otherBullet.id],
-                extra: null,
-              });
+              raiseBulletBullet([o, otherBullet]);
```
Comment (darzu) [inline src/game/bullet-collision.ts:43]: notice how much nicer it is to raise events

File: src/net/events.ts [hunk 2]
```diff
@@ -560,3 +561,48 @@ export function addEventComponents(em: EntityManager) {
...
+      ...(opts?.deserializeExtra
+        ? { deserializeExtra: opts.deserializeExtra }
+        : {}),
+    });
+  };
+
+  if (delayInit) onInit(initThunk);
+  else initThunk();
+
+  const raiseEvent = (es: ESet<ES>, extra?: Extra) => {
```
Comment (dwoos) [inline src/net/events.ts:599]: Could we make this take either objects or IDs? We may sometimes want to raise an event w/o having the entities already (e.g., the `hat-drop` event  in `hat.ts`)

File: src/game/bullet-collision.ts (hunk 1 above)
Comment (dwoos) [reply src/game/bullet-collision.ts:43]: yeah this is sick

File: src/net/events.ts [hunk 3]
```diff
@@ -560,3 +561,48 @@ export function addEventComponents(em: EntityManager) {
   em.addSingletonComponent(DetectedEventsDef);
   em.addSingletonComponent(EventsDef);
 }
+
+export function eventWizard<ES extends EDef<any>[], Extra>(
```
Comment (dwoos) [inline src/net/events.ts:565]: Any reason not to make the 🧙 the only way to register an event handler? We don't actually lose any expressivity, do we?

Comment (darzu) [reply src/net/events.ts:565]: We don't, just didn't want to switch everything unless u think its good too

File: src/net/events.ts (hunk 2 above)
Comment (darzu) [reply src/net/events.ts]: oh hmm should be do-able, let me try

**Pull request merged**